.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,87 @@
+name: Docker
+
+on:
+  push:
+    branches: [ "main" ]
+    paths: Dockerfile
+  pull_request:
+    branches: [ "main" ]
```

Comment:
Not sure about this? I can't find documentation that indicates what this should do? PRs targetting that branch or?